arch/arm/src/stm32h7/Kconfig

```diff
@@ -284,10 +284,24 @@ config STM32H7_HAVE_SMPS
 	bool
 	default n
 
+choice
+	prompt "STM32 H7 Power Supply Selection"
+	default STM32H7_PWR_DIRECT_SMPS_SUPPLY
+	depends on STM32H7_HAVE_SMPS && !HAS_PWR_DIRECT_SMPS_SUPPLY
```

Comment:
where you define HAS_PWR_DIRECT_SMPS_SUPPLY

Reply:
@xiaoxiang781216 

Is HAS_PWR_DIRECT_SMPS_SUPPLY defined in the wrong place?
I'm not sure where to define it correctly.
If it's in the wrong place, please tell me the correct place.

Reply:
All option must be defined through config keyword before using, but where do you define the new option HAS_PWR_DIRECT_SMPS_SUPPLY?

Reply:
I corrected it.